### PR TITLE
Implement pow operator for BigDecimal and BigRational

### LINKED ISSUE #7680: BigDecimal missing ** method
Converting this snippet from Ruby to Crystal.

```
require "bigdecimal"
 
LN2 = BigMath::log(2,16) 

def xx(n)
     ...
     2 * LN2  ** (n + 1)
end
```

Because Crystal has no ``**`` and ``log`` methods, I had to do this, which has less precision.

```
require "big"
include Math

LN2 = Math.log(2)

def xx(n)
     ...
     2 * LN2.to_big_d  ** (n + 1)     # can't do, no ** method for BigDecimal
     2 * LN2.to_big_f  ** (n + 1)     # not as much precision (digits) as Ruby
end
```
BigInt and BigFloat have ``**`` while BigDecimal and BigRational don't.

### PR DESCRIPTION
**Pull request by jwbuiter**
Resolves #7680

Since @ananagame closed his PR, I decided to try to implement this for my first contribution.

Instead of an iterative approach, this implementation uses the operators of the underlying BigInt representation to achieve the same results.

I also added tests and tried to format the styles of the tests to match those in the surrounding file.

### CODE REVIEW

File: spec/std/big/big_decimal_spec.cr [hunk 1]
```diff
@@ -204,6 +204,12 @@ describe BigDecimal do
     (3 / 2.to_big_d).should eq(BigDecimal.new("1.5"))
   end
 
+  it "exponentiates" do
+    result = ("12.34".to_big_d ** 5)
```
Comment (Sija) [inline spec/std/big/big_decimal_spec.cr]:
```suggestion
    result = "12.34".to_big_d ** 5
```

Comment (jwbuiter) [reply spec/std/big/big_decimal_spec.cr]: I added this to the batch commit.

File: src/big/big_rational.cr [hunk 2]
```diff
@@ -170,6 +170,23 @@ struct BigRational < Number
     BigRational.new { |mpq| LibGMP.mpq_neg(mpq, self) }
   end
 
+  # Raises the rational to the *other*th power
+  #
+  # This will raise an exception if rational is 0 and other is negative.
```
Comment (r00ster91) [inline src/big/big_rational.cr]:
```suggestion
  # This will raise `ArgumentError` if rational is 0 and *other* is negative.
```

Comment (jwbuiter) [reply src/big/big_rational.cr]: Actually, it raises a `DivisionByZeroError`

Comment (r00ster91) [reply src/big/big_rational.cr]: edited

**Comment by straight-shoota** [comment]
@jwbuiter could you apply the suggestions? You might've fixed them, but they're not in the branch.

**Comment by straight-shoota** [comment]
Thank you @jwbuiter

**Pull request merged**